usaspending_api/etl/management/commands/check_es_doc_duplication.py

```diff
@@ -0,0 +1,245 @@
+from elasticsearch.helpers import scan
+from math import ceil, floor
+from multiprocessing.pool import ThreadPool
+from threading import Lock, get_ident
+
+
+_log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Check whether an index has duplicate documents on different shards with the same ``_id`` value
+
+    Assumptions:
+        - The duplication is determined by two docs existing with the same value for meta field ``_id``
+        - There is some non-default shard-routing on the index that would allow for two docs with the same ``_id``
+          value to end up on different shards (by default, shard-routing is done using the ``_id`` field)
+    """
+
+    help = """
+        Check whether an index has duplicate documents on different shards with the same ``_id`` value.
+        There are two approaches, and one needs to be chosen using either --query-type scroll, or --query-type 
+        partition. See arg info for --query-type for more info. There is no fast way to perform this check, 
+        but these two strategies provide a client-side-intense way to do it or a server-side intense way to do it, 
+        respectively. Each takes a very long time to process (e.g. 8hrs for 100M docs). Trial out performance and 
+        impact to client or server by testing with a small --stop-after value.
+    """
+
+    _es_client_config: dict = None
+    _index: str = None
+    _partition_size: int = None
+    _num_partitions: int = None
+    _lock = Lock()
+    _completed_count: int = 0
+    _percents = {p for p in range(1, 101)}
+    _duplicated_doc_ids: dict = {}
+    _duplicated_query_template = {
+        "size": 0,
+        "aggs": {
+            "count_duplication_by_partitions": {
+                "terms": {
+                    "field": "_id",
+                    "min_doc_count": 2,
+                    "size": None,  # to be updated
+                    "include": {"partition": None, "num_partitions": None},  # to be updated  # to be updated
+                }
+            }
+        },
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--query-type",
+            type=str,
+            choices=("scroll", "partition"),
+            help="The query strategy to use. scroll: will walk across ALL docs, querying --partition-size results at a "
+            "time. The results are then aggregated in Python on the client-side, so it could be memory intensive"
+            "on the client host on which it is run. partition: will run terms aggregations on disjoint "
+            "partitions of the docs, returning --partition-size agg buckets with each call. The aggs being done "
+            "concurrently on the server can cause server straing on CPU, memory, and induce request timeouts",
+        )
+        parser.add_argument(
+            "--es-hostname",
+            type=str,
+            help="The HTTP(S) URL of the Elasticsearch cluster endpoint",
+        )
```

Comment:
Could use the env var + global variable `ES_HOSTNAME` like the other scripts

Reply:
Left out for flexibility of how this is used - a bit more flexibility by not having to have env vars setup and pulled in via `settings.py` to run this. Could be future improvement if used regularly